### PR TITLE
Check for last response within same api instance

### PR DESCRIPTION
**Pull request by Fredar**
Fixes account synchronization issues

### CODE REVIEW

**Comment by wijet** [comment]
I think we should memoize the api client on the Full strategy object level, then it's safe. For the lifetime of one synchronization object it will use the same client. There are no specs in your example, so I was unable to test it, but something like this should work. 

```
def api
  return @api if @api
  closest = [@scope, @scope.class, @model_class].find do |object|
    object.respond_to?(:api)
   end
  @api ||= closest.try(:api) || raise(MissingAPIClient.new(@scope, @model_class))
end
```
What do you think?

**Comment by Fredar** [comment]
:+1: I was thinking about it as well, wasnt sure if its safe though.

**Comment by Fredar** [comment]
Hmm code in present state is strange:
```
def meta
  remote_objects
  @meta ||= api.last_response.meta
end
```
Triggers this code 
```
def remote_objects
  @remote_objects ||= @perform_request ? fetch_remote_objects : nil
 end
```

So if `@perform_request` evals to `false`, there will be no request and `@meta ||= api.last_response.meta` will call `meta` on `nil`?

**Comment by wijet** [comment]
If @perform_request is false it means that no requst is needed, because all objects are in @remote_objects. This also mean that we are in full strategy and no meta is required.

**Comment by Fredar** [comment]
Hmm ok, this is a bit dirty, but I can see the reasons.

**Comment by wijet** [comment]
Actually, no meta object is defined in this case. Why it's dirty?

**Comment by Fredar** [comment]
I am talking about `updated_since` perspective. Uses a method from `full`, but doesnt need its full logic.

File: lib/synced/strategies/full.rb [hunk 1]
```diff
@@ -147,7 +147,7 @@ def api
         closest = [@scope, @scope.class, @model_class].find do |object|
                     object.respond_to?(:api)
                   end
-        closest.try(:api) || raise(MissingAPIClient.new(@scope, @model_class))
+        @api ||= closest.try(:api) || raise(MissingAPIClient.new(@scope, @model_class))
```
Comment (wijet) [inline lib/synced/strategies/full.rb]: Do we have a test for this case? That if I call this method from an object twice, on second call we will receive exactly the same object as on the first one.

**Comment by wijet** [comment]
:+1:

File: lib/synced/strategies/full.rb (hunk 1 above)
Comment (Fredar) [inline lib/synced/strategies/full.rb]: Tested here https://github.com/BookingSync/synced/pull/37/files#diff-96d1c2199bc62c30543042e083998f57R30

**Pull request merged**